src/main/java/bio/terra/service/dataset/StorageResourceDao.java

```diff
@@ -0,0 +1,97 @@
+    private static final String SQL_GET = "SELECT " + STORAGE_COLUMNS +
+        "FROM storage_resource WHERE dataset_id = :dataset_id";
+    private static final Logger logger = LoggerFactory.getLogger(StorageResourceDao.class);
+    private final NamedParameterJdbcTemplate jdbcTemplate;
+
+    @Autowired
+    public StorageResourceDao(DataRepoJdbcConfiguration jdbcConfiguration) {
+        jdbcTemplate = new NamedParameterJdbcTemplate(jdbcConfiguration.getDataSource());
+    }
+
+    @Transactional(propagation = Propagation.REQUIRED, readOnly = true)
+    public List<StorageResource> getStorageResourcesByDatasetId(UUID datasetId) {
+        try {
+            MapSqlParameterSource params = new MapSqlParameterSource()
+                .addValue("dataset_id", datasetId);
+            return jdbcTemplate.query(SQL_GET, params, new StorageResourceMapper());
+        } catch (EmptyResultDataAccessException ex) {
+            throw new StorageResourceNotFoundException("Storage resource not found for dataset: "
+                + datasetId.toString());
+        }
+    }
+
+    private static class StorageResourceMapper implements RowMapper<StorageResource> {
+        public StorageResource mapRow(ResultSet rs, int rowNum) throws SQLException {
+            return new StorageResource()
+                .cloudPlatform(CloudPlatform.valueOf(rs.getString("cloud_platform")))
+                .cloudResource(GoogleCloudResource.valueOf(rs.getString("cloud_resource")))
+                .region(GoogleRegion.valueOf(rs.getString("region")));
+        }
+    }
+
+    @Transactional(propagation =  Propagation.REQUIRED, isolation = Isolation.SERIALIZABLE)
+    public void createStorageAttributes(List<StorageResource> storageResources, UUID datasetId) {
+        logger.debug("Create Operation: createStorageAttributes datasetId: {}", datasetId);
+
+        MapSqlParameterSource params = new MapSqlParameterSource()
+            .addValue("dataset_id", datasetId);
+
+        List<String> valuesList = new ArrayList<>();
+        for (StorageResource storageResource : storageResources) {
+            String regionParam = storageResource.getCloudResource() + "_region";
+            String cloudResourceParam = storageResource.getCloudResource() + "_cloudResource";
+            String platformParam = storageResource.getCloudResource() + "_cloudPlatform";
+            valuesList.add(String.format("(:dataset_id, :%s, :%s, :%s)",
+                regionParam, cloudResourceParam, platformParam));
+            params.addValue(regionParam, storageResource.getRegion().name());
+            params.addValue(cloudResourceParam, storageResource.getCloudResource().name());
+            params.addValue(platformParam, storageResource.getCloudPlatform().name());
+        }
+
+        String sql = "INSERT INTO storage_resource " +
+            "(dataset_id, region, cloud_resource, cloud_platform) " +
+            "VALUES " + String.join(",\n", valuesList);
```

Comment:
It seems odd to me to add `\n` as this query is not printed.

Reply:
Its useful for debugging. We need `String.join` anyways, so might as well make it pretty.